lib/ClusterShell/RangeSet.py

```diff
@@ -225,7 +253,8 @@ def dim(self):
 
     def _sorted(self):
         """Get sorted list from inner set."""
-        return sorted(set.__iter__(self))
+        # for mixed padding support, sort by both string length and index
+        return sorted(set.__iter__(self), key=lambda x: (len(x), int(x)))
```

Comment:
since we compare length first we don't actually need to cast to int here for sorting: "03" > "01" just the same as 3 > 1 for a given length (might save a tiny bit of time)